e2e/xbridge/token.go

```diff
@@ -0,0 +1,181 @@
+package xbridge
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/netconf"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+type tokenDeploymentConfig struct {
+	Config     deploymentConfig
+	Name       string
+	Symbol     string
+	Minter     common.Address
+	Admin      common.Address
+	Upgrader   common.Address
+	Pauser     common.Address
+	Clawbacker common.Address
+}
+
+func (cfg tokenDeploymentConfig) validateTokenConfig() error {
```

Comment:
```suggestion
func (cfg tokenDeploymentConfig) validate() error {
```

the receive `(cfg tokenDepoloymentConfig)` provides enough context